### DISCUSSION TITLE
framer-motion for Next.js 15.0.2

### DISCUSSION THREAD

**Discussion by NahidKakon** [vercel/next.js#72228] [Help]
### Summary

dear all, I am new in software development, trying to deploy in Vercel. it's not working due to framer motion. I tried it with --force and also with --legacy-peer-deps, didn't work. Downgrading Next.js to 14 and React 18 changes my website's look. So I want to keep Next.js 15, React 19 and use framer-motion, is there any way to solve it?
Thanks in advance for any help or advice

### Additional information

```js
Error log states 
41 error code ERESOLVE
42 error ERESOLVE could not resolve
43 error
44 error While resolving: framer-motion@11.11.11
44 error Found: react@19.0.0-rc-02c0e824-20241028[2m[22m
```

**Comment by dultengmax** [comment]
It seems that it’s still in the adjustment phase for the Next.js update.

**Comment by VedantRaut-sudo** [comment]
Framer Motion is not yet compatible with React 19. While you could downgrade React to version 18 while keeping Next.js 15, this isn't recommended as it may lead to stability issues. I suggest waiting and I hope Framer Motion will release an update that supports React 19🤞

**Reply by NahidKakon** [reply]
thanks for your reply, it reassures me that my codes were right, it is a framer motion's issue. 
Yes, hoping that framer motion will release an update soon

**Comment by himanshu-tyd** [comment]
I don't know why I used latest version nextjs 15 and react 19 now I am stuck

**Comment by vim-diesel** [comment]
you can use "framer-motion": "12.0.0-alpha.2", it seems to be working fine for me, although its only used for the sonner toast package

**Reply by vim-diesel** [reply]
nvm it seems the latest stable version 11 supports react 19 now

**Comment by himanshu-tyd** [comment]
You can just downgrade the version of react 19 to 18 keeping next-js 15 `install` framer-motion stable version i also did that it's worked well

**Reply by VedantRaut-sudo** [reply]
However, you might miss out on React 19's new features. Additionally, some functionality in Next.js 15 might not work properly if it requires React 19 to perform certain tasks.

**Comment by abelMarkos2023** [comment]
I used to have the same problem when I tried to install framer motion with next.js 15 and I solved it with this command

```
npm i framer-motion --legacy-peer-deps
```

**Comment by VinLvy** [comment]
im facing the same problem with my next.js 15.2.3

**Comment by hanpham32** [comment]
Similar issue here. Animation doesn't work. I'm using next.js 15.3.1, react 19, the latest stable framer-motion. I also tried downgrading react to 18 and used "framer-motion": "12.0.0-alpha.2" but that doesn't fix the issue

**Comment by tommyjepsen** [comment]
Changed from
`import { motion, AnimatePresence } from "framer-motion"; `
to 
`import { motion, AnimatePresence } from "motion/react";`

and ran
`npm install motion`

Helped me. :)

**Reply by VedantRaut-sudo** [reply]
This is correct , whats an issue ? @tommyjepsen
